src/components/ChatEntry.js

```diff
@@ -1,22 +1,48 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+  const onLikedButtonClick = () => {
+    const updatedMessage = {
+      id: props.id,
+      sender: props.sender,
+      body: props.body,
+      timeStamp: props.timeStamp,
+      liked: !props.liked,
+    };
+
+    props.updateMessageData(updatedMessage);
+  };
```

Comment:
hmm you are giving this function a lot of editing power when it only needs to update the `liked` property. Think about giving the function access only to `liked`.